tests/test_wait.py

```diff
@@ -45,3 +45,14 @@ def condition():
 
 def test_return_on_timeout():
     assert wait().at_most(100 * MILLISECOND).return_on_timeout().until(lambda: 2 == 3) is False
+
+
+def test_raise_given_condition_invoked_before_at_least():
+    with pytest.raises(ConditionTimeoutError):
+        with assert_done_after(0.4) as c:
+            wait().at_least(0.6).until(lambda: c.done)
+
+
+def test_condition_passed_crossing_at_least_threshold():
```

Comment:
I'm not sure there's a reason for this test

Reply:
fixed